src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -2435,7 +2435,8 @@ else if (!useTnir) {
 
         }
 
-        assert result.equals(Result.SUCCESS) == true;
+        boolean equalSuccess = result.equals(Result.SUCCESS);
```

Comment:
if we do not handle the scenario when result is not Result.SUCCESS and assertion is not enabled, I guess we should remove the assertion in the original code. But the assertion is there already..